pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -524,7 +525,7 @@ public List<FileMetadata> listFilesWithMetadata(URI fileUri, boolean recursive)
             .setIsDirectory(s3Object.key().endsWith(DELIMITER));
         listBuilder.add(fileBuilder.build());
       }
-    });
+    }, null);
```

Comment:
The `PinotFS::listFilesWithMetadata` Javadoc seems to indicate that even this method should include directories in the returned list. Shouldn't we create an appropriate `CommonPrefix` consumer here as well?